mat/solve.go

```diff
@@ -33,61 +27,18 @@ import (
 // If A does not have full rank, a Condition error is returned. See the
 // documentation for Condition for more information.
 func (m *Dense) Solve(a, b Matrix) error {
```

Comment:
Do we want to have a note here that if the underlying `Matrix` in `a` is a `SolveToer` that its `SolveTo` method is used? If not, why do we define `SolveToer`?

Related: We don't AFAICS have a defined notion of a backing matrix — this would be any matrix that is what you get when you have called `untransposeExtract` with it, though it can't be documented with reference to that since it is not exported. I think this would be useful to add (elsewhen) to the section "Type Switching" in the docs.

Reply:
I've added a note about SolveToer, PTAL.

I'll postpone the update of "Type Switching" to later because I don't immediately see an simple way to describe it and don't want to block this PR.